### ISSUE TITLE
Add form_type filter to efilings endpoint

### ISSUE THREAD

**Issue by cnlucas** [fecgov/openFEC#5497]
### What we’re after

A user reached out to apiinfo@fec.gov as they were trying to set up an alert for when candidates from their state file their Form 2's. Currently, this is not possible through our raw data on our efilings endpoint. Users do not currently have the ability to receive these kind of notices through FECNotify. 

I agree with the user that this functionality would be helpful. We should add state and form_type filters to the efilings endpoint. 

### Related ticket(s)
- [ ] https://github.com/fecgov/openFEC/issues/5577

### Action item(s)

(These are the smaller tasks that should happen in order to complete this work)
- [ ] State and form_type filters have been added to the efilings endpoint
- [ ] Once this work is done, we should reach out to the user to notify them that the functionality is available

### Completion criteria

- [ ] form type filter is  added to efilings endpoint
- [ ] follow-up ticket created for state here https://github.com/fecgov/openFEC/issues/5577

**Comment by cnlucas** [comment]
Note for myself to update and close related tickets

**Comment by cnlucas** [comment]
Adding form_type filter in this ticket. I'm opening a new ticket for state as that will take longer
